File: Data Structures & Algorithms/max-water-container/submission-4.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int maximum = 0;
        int l = 0;
        int r = heights.size() - 1;

        while (l < r){
            int length = r - l;
            int area = length * min(heights[l], heights[r]);

            if (area > maximum){
                maximum = area;
            }

            if (heights[l] < heights[r]){
                l++;
            } else{
                r--;
            }
        }
        return maximum;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        minimum = 1000

        while len(nums) > 1:
            l, r, m = 0, len(nums) - 1, len(nums) // 2
            if nums[l] > nums[m]:
                minimum = nums[m] if nums[m] < minimum else minimum
                nums = nums[l:m]
            elif nums[r] < nums[m]:
                minimum = nums[r] if nums[r] < minimum else minimum
                nums = nums[m:r]
            else:
                break
                
            



        return min(minimum, nums[0])

        
                





